portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,21 +44,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
+    //if user hasn't specified:
+    int userChoice = getChoice(request);
+    System.out.println("user choice: ");
+    System.out.println(userChoice);
+
+
+
 
+
+    //goal of this is to add userChoice number of comments to list
+    // --> the error I'm getting is that everytime it displays only 3 comments and creates an "undefined" comment
     List<Comment> UserComments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      String text = (String) entity.getProperty("text");
-      long timestamp = (long) entity.getProperty("timestamp");
 
-      Comment comment = new Comment(id, text, timestamp);
-      UserComments.add(comment);
+    for (Entity entity : results.asIterable()) {//iterate through the comments
+        if(UserComments.size() < userChoice){//if the user comments hasn't exceeded the desired num of comments
+          
+          //get all the properties
+          long id = entity.getKey().getId();
+          String text = (String) entity.getProperty("text");
+          long timestamp = (long) entity.getProperty("timestamp");
+         
+         //create new comments object
+          Comment comment = new Comment(id, text, timestamp);
+         
+         //add to user comments
+          UserComments.add(comment);
```

Comment:
why does it always add an "undefined" comment?

Reply:
Resolved by changing POST to GET on the get comments form